ibm/data_source_ibm_cm_catalog.go

```diff
@@ -0,0 +1,129 @@
+/**
+ * (C) Copyright IBM Corp. 2021.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM/platform-services-go-sdk/catalogmanagementv1"
+)
+
+func dataSourceIBMCmCatalog() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMCmCatalogRead,
+
+		Schema: map[string]*schema.Schema{
+			"catalog_identifier": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID for catalog",
+			},
+			"rev": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Cloudant revision.",
+			},
+			"label": &schema.Schema{
```

Comment:
"label","short_description","catalog_icon_url".. All these are response attributes and no where used in request paload..
So, these attributes should be Computed